### DISCUSSION TITLE
Async test client

### DISCUSSION THREAD

**Discussion by provinzkraut** [litestar-org/litestar#691] [Ideas]
Might be a nice addition to have an async test client. Thoughts?

**Comment by Goldziher** [comment]
Sure, we can easily expose async methods. You want to add this? 

Another question is whether we actually need httpx for this - we might be able to drop it as a dependency as well, but I haven't really dug into what we actually need from it currently to say for sure.

**Reply by provinzkraut** [reply]
Can do. I'll open an issue.

As for dropping httpx, I don't think we should. It's quite substantial, and I think there's value in having a test client that acts like a real-word HTTP-Client (because it is one).

**Comment by infohash** [comment]
Should we optionally allow the user to enable real HTTP request to test route handlers that call external API? Currently, it throws 404 for URL if it's not the `base_url`.

**Reply by Goldziher** [reply]
So, you mean as a redirect?

**Reply by infohash** [reply]
Well, yes. I don't see any other use case of optionally enabling real HTTP transport except for addressing external redirect in tests.

**Comment by kareemmahlees** [comment]
Newcomer here!
Please assign this to me, I will try my best to implement the test client with httpx.

**Reply by provinzkraut** [reply]
Sure, have a go. See the issue: #692.
You need to make comment there so I can assign you.
